### ISSUE TITLE
Concatenating a normal bedtool object with an empty one returns an empty object

### ISSUE THREAD

**Issue by rolfschr** [daler/pybedtools#162]
[Not sure this is expected or a bug]

Concatenating a normal bedtool object with an empty one returns an empty object. My expectation was to get back an bedtool object with the features from the non-empty input.

Tested on pybedtools 0.7.1.

```
>>> a = pybedtools.example_bedtool('a.bed')

>>> print(a)
chr1    1       100     feature1        0       +
chr1    100     200     feature2        0       +
chr1    150     500     feature3        0       -
chr1    900     950     feature4        0       +


>>> c = pybedtools.BedTool(())

>>> print(c)


>>> a.cat(c)
>>> <BedTool(/tmp/pybedtools.2ijU2h.tmp)>

>>> print(a.cat(c))


>>>
```

**Comment by daler** [comment]
Yep, it's a bug, likely introduced in #138. I'll fix and add tests; thanks for reporting this.

**Comment by daler** [comment]
#138 actually had fixed this, and see [0.7.4 changelog](http://daler.github.io/pybedtools/changes.html#changes-in-v0-7-4). Update pybedtools to at least 0.7.4 and it should be fixed, or at least be more explicit about what's happening. Note there's a difference between a BedTool pointing to an empty file and a BedTool initialized with no file:

```python
a = pybedtools.example_bedtool("b.bed")
c = pybedtools.BedTool()
print(a.cat(c))
```
```
ValueError: Can't check filetype or field count -- is one of the files you're merging a 'streaming' BedTool?  If so, use .saveas() to save to file first
```

```python
a = pybedtools.example_bedtool("b.bed")
c = pybedtools.BedTool("", from_string=True)
print(a.cat(c))
```

Gives:

```
chr1	155	200
chr1	800	901
```
